File: WellnessWingman/Models/SleepAnalysisResult.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace HealthHelper.Models;

public class SleepAnalysisResult
{
    [JsonPropertyName("durationHours")]
    public double? DurationHours { get; set; }

    [JsonPropertyName("sleepScore")]
    public double? SleepScore { get; set; }

    [JsonPropertyName("qualitySummary")]
    public string? QualitySummary { get; set; }

    [JsonPropertyName("environmentNotes")]
    public List<string> EnvironmentNotes { get; set; } = new();

    [JsonPropertyName("recommendations")]
    public List<string> Recommendations { get; set; } = new();
}

File: shared/src/commonMain/kotlin/com/wellnesswingman/data/model/analysis/SleepAnalysisResult.kt
package com.wellnesswingman.data.model.analysis

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Structured JSON schema for sleep analysis results from the LLM.
 */
@Serializable
data class SleepAnalysisResult(
    @SerialName("durationHours")
    @Serializable(with = LenientNullableDoubleSerializer::class)
    val durationHours: Double? = null,

    @SerialName("sleepScore")
    @Serializable(with = LenientNullableDoubleSerializer::class)
    val sleepScore: Double? = null,

    @SerialName("qualitySummary")
    val qualitySummary: String? = null,

    @SerialName("environmentNotes")
    val environmentNotes: List<String> = emptyList(),

    @SerialName("recommendations")
    val recommendations: List<String> = emptyList()
)
